src/Records/Types/DomainName.php

```diff
@@ -87,12 +89,8 @@ public function setLabels(array $labels, $tldFirst = false)
         $length = $count = 0;
 
         foreach ($labels as &$label) {
-            $label = \LibDNS\normalize_name($label);
-            $labelLength = \strlen($label);
-            if ($labelLength > 63) {
-                throw new \InvalidArgumentException('Label list is not a valid domain name: Label ' . $label . ' length exceeds 63 byte limit');
-            }
-            $length += $labelLength + 1;
+            $label = (string)new NormalizedLabel($label);
```

Comment:
True, but I rather see value objects than public static functions. A value object *can* be passed around, while public static functions can't. Meaning: an object is ready for change while a public static function is not.

Hence, I avoid public static functions at any cost. For PHP I only see value in public static functions as named constructors.